lib/Doctrine/Common/Collections/Criteria.php

```diff
@@ -109,6 +109,12 @@ public function __construct(Expression $expression = null, array $orderings = nu
         }
     }
 
+    public static function fromArray(array $criteria) {
```

Comment:
this method was added to allow one-liner creation, as ``(new Criteria())->orderBy(array('foo' => 'DESC'))`` does not work on PHP 5.3, which is the min requirement of the library.

When building the whole criteria as a nested array externally and then converting it to a doctrine criteria, I don't expect wanting to continue changing it inline. And I would prefer keeping this criteria converter separate (I'm not sure I would want it in core, but I'm not the one making the decision anyway)